core/src/main/java/hudson/Functions.java

```diff
@@ -1903,6 +1905,23 @@ public static String joinPath(String... components) {
             return joinPath(Stapler.getCurrentRequest().getContextPath() + '/' + itUrl, urlName);
     }
 
+    /**
+     * Computes the link to the console for the run for the specified executable, taking {@link ConsoleUrlProvider} into account.
+     * @param executable the executable (normally a {@link Run})
+     * @return the absolute URL for accessing the build console for the executable, or null if there is no build associated with the executable
+     * @since TODO
+     */
+    public static @CheckForNull String getConsoleUrl(Queue.Executable executable) {
+        if (executable == null) {
+            return null;
+        } else if (executable instanceof Run) {
+            return ConsoleUrlProvider.getRedirectUrl((Run<?, ?>) executable);
```

Comment:
I was thinking about this, and I could see an argument for changing `ConsoleUrlProvider` to work with `Queue.Executable` directly as well. For example, branch indexing and org folder scans use `FolderComputation`, which is a `Queue.Executable` but has no associated `Run`. I am somewhat doubtful that anyone will want to create external visualizations for those logs, but maybe if we are introducing a new API anyway we should make it as generic as possible? The tradeoff is additional complexity in each individual extension, but maybe we could add a `ConsoleUrlProvider.ForRun` subclass that most implementations could use and only handles `Run`, and then anyone who really needs to do something special for `Queue.Executable` could implement `ConsoleUrlProvider` directly?